### PR TITLE
Simplify Input Table Interface.

### PR DESCRIPTION
**Pull request by cpwright**
Removes the InputTableEnumGetter and InputTableRowSetter interfaces that MutableInputTable extended, but are not used by web.

There have been the following class renames:
io.deephaven.engine.util.config.MutableInputTable -> io.deephaven.engine.util.input.InputTableHandler
KeyedArrayBackedMutableTable -> io.deephaven.engine.table.impl.util.KeyedArrayBackedInputTable
AppendOnlyArrayBackedMutableTable -> io.deephaven.engine.table.impl.util.AppendOnlyArrayBackedInputTable

Additionally, the MutableInputTable interface, KeyedArrayBackedMutableTable, and AppendOnlyArrayBackedMutableTable no longer have parameters for enum values.  The "allowEdits" parameter has been removed from the interface and all implementations.  The delete method that takes a rowSet has been removed.

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/util/config/MutableInputTable.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
  * Implementations of this interface will make their own guarantees about how atomically changes will be applied and
  * what operations they support.
  */
-public interface MutableInputTable extends InputTableRowSetter, InputTableEnumGetter {
+public interface MutableInputTable {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/util/config/MutableInputTable.java:23]: While you're in this code, can you change the interface from `RowSet` to `TrackingRowSet`?

Also, should we be adding an `add` overload for `allowEdits`? Should we rename that boolean `allowModify`?

File: engine/table/src/main/java/io/deephaven/engine/table/impl/util/AppendOnlyArrayBackedMutableTable.java [hunk 2]
```diff
@@ -88,7 +87,7 @@ private AppendOnlyArrayBackedMutableTable(@NotNull TableDefinition definition,
             final Map<String, Object[]> enumValues, final ProcessPendingUpdater processPendingUpdater) {
         // noinspection resource
         super(RowSetFactory.empty().toTracking(), makeColumnSourceMap(definition),
-                enumValues, processPendingUpdater);
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/util/AppendOnlyArrayBackedMutableTable.java:91]: We still have other constructors accepting this parameter, and now we ignore it...

File: engine/table/src/main/java/io/deephaven/engine/table/impl/util/BaseArrayBackedMutableTable.java [hunk 3]
```diff
@@ -45,18 +39,15 @@ abstract class BaseArrayBackedMutableTable extends UpdatableTable {
...
-    private final Map<String, Object[]> enumValues;
-
     private String description = getDefaultDescription();
     private Runnable onPendingChange = updateGraph::requestRefresh;
 
     long nextRow = 0;
     private long pendingProcessed = -1L;
 
     public BaseArrayBackedMutableTable(TrackingRowSet rowSet, Map<String, ? extends ColumnSource<?>> nameToColumnSource,
-            Map<String, Object[]> enumValues, ProcessPendingUpdater processPendingUpdater) {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/util/BaseArrayBackedMutableTable.java:57]: Unused `enumValues` in other constructors, as well.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/util/KeyedArrayBackedMutableTable.java [hunk 4]
```diff
@@ -110,7 +110,7 @@ private KeyedArrayBackedMutableTable(@NotNull TableDefinition definition, final
             final Map<String, Object[]> enumValues, final ProcessPendingUpdater processPendingUpdater) {
         // noinspection resource
         super(RowSetFactory.empty().toTracking(), makeColumnSourceMap(definition),
-                enumValues, processPendingUpdater);
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/util/KeyedArrayBackedMutableTable.java:113]: Same trouble here.

File: engine/table/src/main/java/io/deephaven/engine/util/config/InputTable.java [hunk 5]
```diff
@@ -15,12 +15,12 @@
...
 /**
  * A minimal interface for mutable shared tables, providing the ability to write to the table instance this is attached
- * to. MutableInputTable instances are set on the table as an attribute.
+ * to. InputTable instances are set on the table as an attribute.
  * <p>
  * Implementations of this interface will make their own guarantees about how atomically changes will be applied and
  * what operations they support.
  */
-public interface MutableInputTable {
+public interface InputTable {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/util/config/InputTable.java]: `InputTableHandler`?

**Review by rcaudy — APPROVED** [review]

**Review by jmao-denver — APPROVED** [review]
Python changes LGTM

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

How-to: https://github.com/deephaven/deephaven.io/issues/3528
Reference: https://github.com/deephaven/deephaven.io/issues/3529

**Pull request merged**